teuthology/provision.py

```diff
@@ -29,105 +30,200 @@ def _get_downburst_exec():
+            log.error("Error destroying {machine}: {msg}".format(
+                machine=self.name, msg=err))
+            return False
+        elif proc.returncode == 0:
+            out_str = ': %s' % out if out else ''
+            log.info("Destroyed %s%s" % (self.name, out_str))
+            return True
+        else:
+            log.error("I don't know if the destroy of {node} succeded!".format(
+                node=self.name))
+            return False
+
+    def build_config(self):
+        """
+        Assemble a configuration to pass to downburst, and write it to a file.
+        """
+        config_fd = tempfile.NamedTemporaryFile(delete=False)
 
         file_info = {
             'disk-size': '100G',
             'ram': '1.9G',
             'cpus': 1,
             'networks': [
-                {'source': 'front', 'mac': status_info['mac_address']}],
-            'distro': os_type.lower(),
-            'distroversion': os_version,
+                {'source': 'front', 'mac': self.status['mac_address']}],
+            'distro': self.os_type.lower(),
+            'distroversion': self.os_version,
             'additional-disks': 3,
             'additional-disks-size': '200G',
             'arch': 'x86_64',
         }
-        fqdn = machine_name.split('@')[1]
+        fqdn = self.name.split('@')[1]
         file_out = {'downburst': file_info, 'local-hostname': fqdn}
-        yaml.safe_dump(file_out, tmp)
-        metadata = "--meta-data=%s" % tmp.name
-        dbrst = _get_downburst_exec()
-        if not dbrst:
-            log.error("No downburst executable found.")
-            return False
-        p = subprocess.Popen([dbrst, '-c', phys_host,
-                              'create', metadata, createMe],
-                             stdout=subprocess.PIPE, stderr=subprocess.PIPE,)
-        owt, err = p.communicate()
-        if err:
-            log.info("Downburst completed on %s: %s" %
-                     (machine_name, err))
-        else:
-            log.info("%s created: %s" % (machine_name, owt))
-        # If the guest already exists first destroy then re-create:
-        if 'exists' in err:
-            log.info("Guest files exist. Re-creating guest: %s" %
-                     (machine_name))
-            destroy_if_vm(ctx, machine_name)
-            create_if_vm(ctx, machine_name)
-    return True
+        yaml.safe_dump(file_out, config_fd)
+        self.config_path = config_fd.name
+        return True
+
+    def remove_config(self):
+        """
+        Remove the downburst configuration file created by build_config()
+        """
+        if self.config_path and os.path.exists(self.config_path):
+            os.remove(self.config_path)
+            self.config_path = None
+            return True
+        return False
+
+    def __del__(self):
+        self.remove_config()
 
 
-def destroy_if_vm(ctx, machine_name, user=None, description=None):
+def create_if_vm(ctx, machine_name, _downburst=None):
```

Comment:
Would you please mention in the comments that the _downburst kwarg is used only for testing? The destroy method needs this as well.